File: engine/transpile/src/test/resources/golden/SYK004/LK_要求数量.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK004 のデータ部レコード LK-要求数量 を逐語対訳した自動生成コード。 */
public final class LK_要求数量 {

    /** 01 LK-要求数量 総バイト長 3。 */
    public static final int LENGTH = 3;

    private final byte[] data;

    public LK_要求数量() {
        this.data = new byte[3];
    }

    public LK_要求数量(byte[] source) {
        this.data = Arrays.copyOf(source, 3);
    }

    public byte[] data() {
        return data;
    }

    // LK-要求数量
    public long get_LK_要求数量() {
        return CobolRuntime.decodePacked(data, 0, 3);
    }
    public void set_LK_要求数量(long value) {
        CobolRuntime.encodePacked(data, 0, 3, true, value);
    }
}

File: engine/transpile/src/test/resources/golden/SYK004/LK_要求数量.py
"""SYK004 のデータ部レコード LK-要求数量 を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class LK_要求数量:
    """01 LK-要求数量 総バイト長 3。"""

    LENGTH = 3

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(3)
        else:
            self._data = bytearray(data)
            if len(self._data) < 3:
                self._data.extend(b"\x00" * (3 - len(self._data)))

    @property
    def data(self):
        return self._data

    # LK-要求数量
    def get_LK_要求数量(self):
        return decode_packed(self._data, 0, 3)
    def set_LK_要求数量(self, value):
        encode_packed(self._data, 0, 3, True, value)
